File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.java
class Solution {
    public int kthSmallest(int[][] matrix, int k) {
        PriorityQueue<Integer> pq = 
            new PriorityQueue<Integer>(Collections.reverseOrder()); // max heap 
        
        
        for(int i = 0; i < matrix.length; i++){
            for(int j = 0; j < matrix[0].length; j++){
                pq.add(matrix[i][j]);
    
                while(pq.size() > k){
                    pq.poll();
                }
            
            }
        }
        
        return pq.peek();
    }
}

File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.cpp
class Solution {
public:
    int kthSmallest(vector<vector<int>>& matrix, int k) {
        int n = matrix.size();
        priority_queue<int> pq;
        
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                while(pq.size() > k) pq.pop();
                pq.push(matrix[i][j]);
            }
        }
        while(pq.size() > k) pq.pop();
        return pq.top();
    }
};